### ISSUE TITLE
Change triggers of 'view_cart_event'

### ISSUE THREAD

**Issue by koentjeh** [yireo/Yireo_GoogleTagManager2#139]
Version 3.1.2 contains 2 triggers for the `view_cart` event. One in `checkout_cart_index.xml` which seems correct and one triggering after a product is added to cart which now gives unexpected behaviour. It triggers the `view_cart` event when adding a product to cart without the cart being viewed/opened. This is fixed by adding configuration that restricts pushing `view_cart` event in the `checkout/cart` section.
```xml
<type name="Magento\Checkout\CustomerData\Cart">
    <plugin name="Yireo_GoogleTagManager2::addAdditionalDataToCartSection" type="Yireo\GoogleTagManager2\Plugin\AddDataToCartSection"/>
</type>
```

I was thinking about changing the triggers from `addAdditionalDataToCartSection` to Javascript onclick events on specific items, with configuration option on when to trigger. Minicart click, view cart button click.
![Screenshot 2023-04-10 at 11 28 38](https://user-images.githubusercontent.com/6535718/230876111-4da38909-535e-460f-960b-c4cb3954f616.png)

Is there any specific reason why it is done this way and not via a Javascript onclick event on the minicart or view cart button? Performance, prevent waiting for Javascript to load, e.g.?

**Comment by jissereitsma** [comment]
I personally consider it kind of a `view_cart` event if the minicart is showing the number of items in the cart, even though the rest of the minicart is expanded. But I get your point (and Bart mentioned this already as well). My proposal is actually to generate some kind of custom on-click event for the minicart and then have the module push the data. Still, this means that the event data will be generated PHP-side like it is now, but only with a new meta parameter `event` or so. This is work in progress for next week or so.

**Comment by koentjeh** [comment]
We could ask the marketing companies for you and maybe we'll figure something out to make this nicely configurable so that customers would have more control of its behaviour.

I hope to get a response within this or next week. I'll let you know.

**Comment by koentjeh** [comment]
The response was that its advised to trigger the `view_cart` event on the checkout page.

The `view_cart` event is used to analyse the customer journey. E.g. the process of how the customer reached the website until the purchase. The right fit will be to identify when the customer reached the checkout. Not when the customer has any visual feedback of the contents of the cart, such as the amount in the cart or clicking on the mini cart.

This means that the `onclick` event will be unnecessary. Though, i'm willing to help to implement the ability to configure which triggers will be active.

**Comment by jissereitsma** [comment]
@amjadm61 Good catch. This will be fixed in the upcoming release: https://github.com/yireo/Yireo_GoogleTagManager2/commit/2e50b852f24f0aaac894662c54115134e645d123
